src/hooks/useTabs.ts

```diff
@@ -69,7 +69,11 @@ export const useTabs: UseStore<TabsState> = create((set, get) => ({
     if (tab.tabId === activeTab.tabId && tabIndex !== -1) {
       const tabToTheRight = updatedTabs[tabIndex];
       const lastOpenTab = updatedTabs[updatedTabs.length - 1];
-      set({ activeTab: tabToTheRight || lastOpenTab });
+      const newActiveTab = tabToTheRight ?? lastOpenTab;
+      if (newActiveTab) {
+        set({ activeTab: newActiveTab });
```

Comment:
I think we still need to set the `activeTab` even if the `newActiveTab` is undefined. Otherwise `activeTab` would still point to the previously closed tab. We just don't call `onActivate()` if `newActiveTab` is undefined.

Reply:
Oh yea good point thank you